### PR TITLE
fix: #56

### PR DESCRIPTION
**Pull request by nitedani**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/query/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #56 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by nitedani** [comment]
My VS-code markdown auto-format is going crazy. 😄

**Comment by nitedani** [comment]
SSR is working as expected: [example](https://github.com/nitedani/vite-plugin-angular/blob/main/examples/vps-express-telefunc/renderer/_default.page.server.ts)

**Comment by NetanelBasal** [comment]
Can you explain the PR, please? why do we need this?

**Comment by nitedani** [comment]
In order for the online and window focus events to work , [mount](https://github.com/TanStack/query/blob/899773a48681f363efcbf53ce0818e0d393d583f/packages/query-core/src/queryClient.ts#L82) and [unmount](https://github.com/TanStack/query/blob/899773a48681f363efcbf53ce0818e0d393d583f/packages/query-core/src/queryClient.ts#L97) need to be called.
react-query calls it [here](https://github.com/TanStack/query/blob/899773a48681f363efcbf53ce0818e0d393d583f/packages/react-query/src/QueryClientProvider.tsx#L77)
solid-query calls it [here](https://github.com/TanStack/query/blob/899773a48681f363efcbf53ce0818e0d393d583f/packages/solid-query/src/QueryClientProvider.tsx#L84)
vue-query calls it [here](https://github.com/TanStack/query/blob/899773a48681f363efcbf53ce0818e0d393d583f/packages/vue-query/src/vueQueryPlugin.ts#L59)

This PR adds the missing piece of code that those integrations have.

**Comment by NetanelBasal** [comment]
Remove the new token and use the `PLATFORM_ID` to see if we need to run the mount functionality.

**Comment by nitedani** [comment]
How about now? I removed the export of the new token. I would like have a way to provide my own query client, on which the mount and unmount methods will be called, this is why I haven't removed it entirely.

File: packages/ng-query/src/lib/query-client.ts [hunk 1]
```diff
@@ -1,13 +1,55 @@
...
+  Injectable,
+  InjectionToken,
+  OnDestroy,
+  PLATFORM_ID,
+  Provider,
+} from '@angular/core';
 import { QueryClient as QueryCore } from '@tanstack/query-core';
 import { QUERY_CLIENT_OPTIONS } from './providers';
 
+const QueryClient = new InjectionToken<QueryCore>('QueryClient', {
```
Comment (NetanelBasal) [inline packages/ng-query/src/lib/query-client.ts:13]: Remove this

File: packages/ng-query/src/lib/query-client.ts [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+class QueryClientHooks implements OnDestroy {
+  instance = inject(QueryClient);
+  platformId = inject(PLATFORM_ID);
+
+  constructor() {
+    if (isPlatformBrowser(this.platformId)) {
```
Comment (NetanelBasal) [inline packages/ng-query/src/lib/query-client.ts:35]: move to the QueryClientService and call it once. Expose an `init` method so you that you can call it only if in client mode

File: packages/ng-query/src/lib/query-client.ts [hunk 3]
```diff
@@ -1,13 +1,55 @@
...
 import { QUERY_CLIENT_OPTIONS } from './providers';
 
+const QueryClient = new InjectionToken<QueryCore>('QueryClient', {
+  providedIn: 'root',
+  factory() {
+    return new QueryCore(inject(QUERY_CLIENT_OPTIONS));
+  },
+});
+
+export const provideQueryClient = (queryClient: QueryCore): Provider => {
```
Comment (NetanelBasal) [inline packages/ng-query/src/lib/query-client.ts:20]: why we need this?

**Comment by nitedani** [comment]
How about now? We need `provideQueryClient` for situations where we need control over the `QueryClient` instance outside of angular. I added such an example of SSR usage in the readme, where `provideQueryClient` is used.

File: packages/ng-query/src/lib/query-client.ts [hunk 4]
```diff
@@ -1,13 +1,55 @@
...
+  }
+}
+
 export const QueryClientService = new InjectionToken<QueryCore>(
   'QueryClientService',
   {
     providedIn: 'root',
     factory() {
-      return new QueryCore(inject(QUERY_CLIENT_OPTIONS));
+      inject(QueryClientHooks);
```
Comment (NetanelBasal) [inline packages/ng-query/src/lib/query-client.ts:51]:
```ts
const hooks = inject(QueryClientHooks);
if( isPlatformBrowser(inject(PLATFORM_ID))) {
  hooks.init()
}
```

Comment (nitedani) [reply packages/ng-query/src/lib/query-client.ts:51]:
```ts
@Injectable({
  providedIn: 'root',
})
class QueryClientHooks implements OnDestroy {
  instance = inject(QueryClient);
  initialized = false;

  init() {
    this.instance.mount();
    this.initialized = true;
  }

  ngOnDestroy() {
    if (this.initialized) {
      this.instance.unmount();
      this.initialized = false;
    }
  }
}
```
Would this be ok?

Comment (NetanelBasal) [reply packages/ng-query/src/lib/query-client.ts:51]: why we need the check this.initialized

Comment (nitedani) [reply packages/ng-query/src/lib/query-client.ts:51]: Calling `unmount` on an unmounted instance could result in undefined behavior, just like calling `mount` without the `isPlatformBrowser` check could.

**Pull request merged**